libsolidity/formal/SMTChecker.cpp

```diff
@@ -702,6 +704,22 @@ void SMTChecker::resetVariables(vector<Declaration const*> _variables)
 	}
 }
 
+void SMTChecker::mergeVariables(vector<Declaration const*> const& _variables, smt::Expression const& _condition, VariableSequenceCounters const& _countersEndTrue, VariableSequenceCounters const& _countersEndFalse)
+{
+	set<Declaration const*> uniqueVars(_variables.begin(), _variables.end());
+	for (auto const* decl: uniqueVars)
```

Comment:
Another way to do this:
`for (auto const* decl: _variables | boost::adaptors::uniqued) { ... }`
:)

(but no need to change this now)

Reply:
That looks cool :)
But wouldn't that remove only adjacent duplicates? I think in the current code the duplicates are quite far apart.